src/ipfs.rs

```diff
@@ -1,23 +1,20 @@
 use std::{sync::Arc, time::Duration};
 
-use anyhow::Context;
+use anyhow::{anyhow, Context};
 use backoff::{future::retry, ExponentialBackoffBuilder};
-use reqwest::Method;
+use reqwest::{Body, Method};
+use serde::Deserialize;
 
 use crate::{http_client::HttpClient, specification::Specification};
 
-// #[derive(Deserialize)]
-// struct CARUploadResponse {
-//     cid: String,
-// }
-
 pub async fn fetch_specification_with_retry(
     ipfs_http_client: Arc<HttpClient>,
     cid: &String,
 ) -> anyhow::Result<Specification> {
     let fetch = || async {
         let response = match ipfs_http_client
-            .request(Method::POST, format!("/api/v0/cat?arg={cid}"))?
+            .request(Method::POST, format!("/api/v0/cat?arg={cid}"))
+            .await?
```

Comment:
Why do we need an `await?` here?

Reply:
The new rate limiter feature (if present in the HTTP client) is applied in this request method so in case we need to wait we need the extra `await`.